src/web/js/dashboard/GoogleAPI.js

```diff
@@ -93,31 +110,449 @@ class GoogleAPI {
+    });
+  }
+
+// function to copy files
+  copyFile = (parentID, sourceFileID, destinationFileName) => {
+    return window.gapi.client.drive.files.copy({
+      'fileId': sourceFileID,
+      'parents' : [ parentID ],
+      'name': destinationFileName,
+    });
+  }
+
+  // function that initiates the create and distribute assignment process
+  createAndDistributeAssignment = (classID, assignmentName, assignmentFileID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      var assignmentInfo = {
+        id: data.nextAssignmentID,
+        name: assignmentName,
+        class: classID,
+        docID: assignmentFileID,
+        opened: [], //list of studentIDs
+        submitted: [], //list of studentIDs
+      };
+      data.assignmentList[data.nextAssignmentID] = assignmentInfo;
+      data.classList[classID].assignments.push(data.nextAssignmentID);
+      data.nextAssignmentID += 1;
+      return this.savePyretData(data).then(() => {
+        return this.getAppFolderID("pyret").then((greatGrandParent) => {
+          var greatGrandParentID = JSON.parse(greatGrandParent.body).files[0].id;
+          var assignmentFolderName = data.classList[classID].name + "_Assignment_" + assignmentName;
+          return this.createAssignmentFolder(greatGrandParentID, assignmentFolderName, assignmentName, classID, assignmentFileID);
+        });
+      });
+    });
+  }
+
+  // function to create the folder for teacher to contain all the assignments
+  createAssignmentFolder = (greatGrandParentID, assignmentFolderName, assignmentName, classID, assignmentFileID) => {
+    return this.createFolder(greatGrandParentID, assignmentFolderName).then((grandParent) => {
+      var grandParentID = grandParent["result"]["id"];
+      return this.createAssignmentsForClass(classID, grandParentID, assignmentName, assignmentFileID);
+    });
+  }
+
+  // function to obtain students in class and initiate the process of creating individual assignment folder with assignment copy
+  createAssignmentsForClass = (classID, grandParentID, assignmentName, assignmentFileID) => {
+    return this.getStudentsInClass(classID).then((studentList) => {
+      for (let s of studentList) {
+        this.createStudentFolderAndAssignment(s, grandParentID, assignmentName, assignmentFileID, classID);
+      }
+    });
+  }
+
+  // function to create an individual folder and assignment copy for each student in class
+  createStudentFolderAndAssignment = (s, grandParentID, assignmentName, assignmentFileID, classID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      var studentAssignmentFolderName = data.classList[classID].name + "_" + assignmentName + "_" + s["lastName"] + "_" + s["firstName"];
+      var studentAssignmentFileName = assignmentName + "_" + s["lastName"] + "_" + s["firstName"] + ".arr";
+      return this.createFolder(grandParentID, studentAssignmentFolderName).then((parent) => {
```

Comment:
Possibly make a button to redistribute assignment & then check which folders exist